### ISSUE TITLE
Is there any maximum number of precede task?

### ISSUE THREAD

**Issue by sleepyeye** [cpp-taskflow/cpp-taskflow#87]
Hello. First of all, I sorry for didn't follow the issue template. I'm not sure whether this is bug or my mistakes. So I cloudn't settle down to the template.

Currently I'm building  high performance graphics application (ray tracer)
and it needs to compute HUGE amount of same function(with different parameters) parallelly.

However when I try to lunch tasks parallelly, sometimes taskflow failed to build task graph.
The problematic code is in below.
```cpp
  tf::Task taskStart = tf.emplace([](){ syncLog("Start  \n"); });
  tf::Task taskEnd = tf.emplace([](){ syncLog("End  \n"); });
  for(int i = 0; i < numSamples ; ++i) {
    tf::Task task = tf.emplace([i](){ /* do something */ });
    taskStart.precede(task);
    task.precede(taskEnd);
  }
...
  tf.wait_for_all();
```
Basically this is parallel for. 
However, when I increased `numSamples` really high, it failed to construct taskflow with these error
`terminated by signal SIGSEGV (Address boundary error`.
When I set `numSamples=1280*720*4` it works. But when I set `numSamples=1280*720*1024` it failed.

I'm suspecting that a task can have limited number of successor tasks and my application some how over it. 

Is there anything that I'm missing?

**Comment by clin99** [comment]
Hi @sleepyeye , thanks for reporting this issue. 

Because you create too many `numSamples=1280*720*1024=943718400` tasks, this will consume too much memory and result in the memory error. Since you are doing a parallel for, I suggest you let a task handle more iterations (e.g. 1024) to reduce the number of tasks.  

Please let me know if this does not work for you.

Thanks.

**Comment by sleepyeye** [comment]
Hi @clin99 , thanks for your reply.

I've got additional question.

As you said, if the error was caused by too many tasks, then is there any magic number which describes the upper limit of the number of tasks?
To me this is important since each of my tasks can also be parallelized so that the number of task may grow exponentially. Therefore, without the specific number, I should sacrifice potential performance for stability.

Thanks.

**Comment by clin99** [comment]
Hi @sleepyeye , there is no fix upper limit on the number of tasks, this all depends on the memory size of your machine. My suggestion is to let a task do more work. Because there is an overhead of task scheduling and the overhead might offset the performance gain from parallelization if your task is very lightweight. Also you should create more tasks than the number of cores on your machine and try to make tasks have equal workload.

For the second question, if you want to let a task spawn a task graph, then what you need is subflow.
Here is the tutorial of subflow: [https://cpp-taskflow.github.io/cpp-taskflow/chapter5.html](url)

Feel free to let me know if you have any question.

Thanks.

**Comment by sleepyeye** [comment]
Hi @clin99 , thanks for your comments.
Based on your comment, I have tested performance of my renderer in quite a lot of setup. 
And I realized that my initial task was too small.
So I grouped tasks and I've got quite a lot of performance improvement.

Thanks!
